spec/avram/instrumentation_spec.cr

```diff
@@ -2,7 +2,8 @@ require "../spec_helper"
 
 describe "Instrumentation" do
   it "publishes the query and args" do
-    TestDatabase.query "SELECT * FROM users"
+    # using block form to make sure the ResultSet is closed
+    TestDatabase.query "SELECT * FROM users" { }
```

Comment:
This caused an error when rolling back the transaction. Surprised this didn't come up before.

Reply:
Wouldn't this use https://github.com/crystal-lang/crystal-db/blob/5a7d27e0c58510458af3894783649d53529753cc/src/db/query_methods.cr#L45 without the block? That method is defined https://github.com/luckyframework/avram/blob/9b184d4b586d783b096a5d377eafd98c78e32026/src/avram/database.cr#L77

Reply:
Yes, but check out the comment above the method. It returns a `ResultSet` that must be closed (which is why I'm surprised it didn't cause issues before this). Because we weren't closing the result set, an error occurred when rolling back the transaction because there was data still in the pipe left to parse. None of that is official lingo, just how I understand it.

Reply:
oh, I see. Sort of like doing `File.open` without the block... Gotcha.